src/shared/components/query/GenesetsJsTree.tsx

```diff
@@ -0,0 +1,216 @@
+import React from 'react';
+import $ from 'jquery';
+import {getHierarchyData} from "shared/lib/StoreUtils";
+//Import jstree.min and style for jstree to work
+import 'jstree/dist/jstree.min'; // tslint:disable-line
+import 'shared/components/query/styles/jstree/style.css'; // tslint:disable-line
+import * as _ from "lodash";
+import { remoteData } from "shared/api/remoteData";
+import CBioPortalAPIInternal, { GenesetHierarchyInfo } from "shared/api/generated/CBioPortalAPIInternal";
+import { observer } from "mobx-react";
+import { observable, ObservableMap } from "mobx";
+import LoadingIndicator from "shared/components/loadingIndicator/LoadingIndicator";
+
+export interface GenesetsJsTreeProps
+{
+    initialSelection: string[];
+    scoreThreshold: string;
+    pvalueThreshold: string;
+    percentile: string;
+    gsvaProfile: string;
+    sampleListId: string|undefined;
+    searchValue: string;
+    onSelect: (map_geneSet_selected:ObservableMap<boolean>) => void;
+}
+
+type JSNode = {
+    original: {description: string, geneset: boolean};
+};
+
+@observer
+export default class GenesetsJsTree extends React.Component<GenesetsJsTreeProps, {}>
+{
+    tree: Element|null;
+    @observable isLoading = true;
+    promisedTree: Promise<Element>;
+    private readonly map_geneSets_selected = new ObservableMap<boolean>();
+
+    constructor(props: GenesetsJsTreeProps)
+    {
+        super(props);
+        this.map_geneSets_selected.replace(props.initialSelection.map(geneSet => [geneSet, true]));
+    }
+    
+    componentDidMount(){
+        this.promisedTree = this.initTree(this.tree as Element); //React only sets the div to null when the component unmounts
+        this.promisedTree.then(tree => (this.isLoading = false, tree));
+    }
+    
+    componentDidUpdate(prevProps: GenesetsJsTreeProps){
+        if (this.props.scoreThreshold !== prevProps.scoreThreshold ||
+            this.props.pvalueThreshold !== prevProps.pvalueThreshold ||
+            this.props.percentile !== prevProps.percentile) {
+            this.promisedTree = this.promisedTree
+                .then(tree => (this.isLoading = true, this.replaceTree(tree)))
+                .then(tree => (this.isLoading = false, tree));
+        }
+        if (this.props.searchValue !== prevProps.searchValue) {
+            this.promisedTree = this.promisedTree
+                .then(tree => this.searchTree(tree, this.props.searchValue));
+        }
+    }
+    
+    /**
```

Comment:
i think we can get rid of these comments since ts annotation should provide the same thing.